### DISCUSSION TITLE
Support for managing custom order types

### DISCUSSION THREAD

**Discussion by NTShop** [woocommerce/woocommerce#35218] [WooCommerce HPOS]
Hi, we need a way to somehow display custom order types (added via wc_register_order_type() ) when the site is using HPOS. Ideally we'd be able to hook in to tell WC to show "this_custom_order_type" instead of "shop_order", and in such a way that all order viewing/editing functionality works as expected including the display order statuses when showing the list of orders, editing a "custom order type", adding a new "custom order type", etc. All this works with the posts table, but currently there's no way to address that with the HPOS/COT implemented on the site.

**Comment by shaybanshee** [comment]
@vedanshujain pinging you in for this one!

**Comment by vedanshujain** [comment]
Thanks, we are scoping currently on how to best support this.

**Comment by barryhughes** [comment]
@NTShop we're tracking this effort [over here](https://github.com/woocommerce/woocommerce/issues/35601). Please feel free to add any notes you think might be helpful. Additionally, I wondered if you would be open to sharing (privately, if you prefer) the code/extension you maintain that requires this capability, so we can use it as a reference point?

**Comment by NTShop** [comment]
@barryhughes I can't share the code, it belongs to a customer and they said not to share, not even privately. However I can explain what is being done in general.

- Register a custom order type, for example "custom_shop_order", using wc_register_order_type( 'custom_shop_order', array() ) where array() is the usual labels etc, with a capability setting of "shop_order".

- Add a new "child order" order using that type, ensure that the parent ID is set to the original WC Order ID.

- When viewing those "child orders" they appear like a regular WC order does, and when viewing/editing one the info also appears like regular WC orders do with all the same content, except of course the products/totals are different because the original WC order is split in new "custom_shop_orders" as "child orders" for the internal purposes of the customer who the code was built for.

- Because WC is currently registering a custom post type somewhere along the line after calling wc_register_order_type() an admin menu item is automatically created under the WooCommerce menu. So that needs to continue somehow or other. 

This is, in a general sense, all very similar to how WC Subscriptions works right now when it creates a subscription, with some differences of course.

**Reply by barryhughes** [reply]
Thanks, I imagined it might be very similar to Subscriptions—but wanted to check before we start laying down code, in case of any notable differences. I appreciate you sharing these notes :-)

**Reply by NTShop** [reply]
@barryhughes - Any progress on this?

**Reply by barryhughes** [reply]
Yes, linked issue https://github.com/woocommerce/woocommerce/issues/35601 is now closed after some work was done in relation to this. Specifically: https://github.com/woocommerce/woocommerce/pull/35658.

**Reply by NTShop** [reply]
@barryhughes  - Ok thanks. Will that new code be in v7.3 or which?

**Reply by barryhughes** [reply]
Yes, exactly—it's milestoned for 7.3.0 🙂
